openassessment/data.py

```diff
@@ -1577,3 +1579,122 @@ def get_file_uploads(self, missing_blank=False):
+
+def parts_summary(assessment_obj):
+    """
+    Retrieves a summary of the parts from a given assessment object.
+    """
+    return [
+        {
+            'type': part.criterion.name,
+            'score': part.points_earned,
+            'score_type': part.option.name if part.option else "None",
+        }
+        for part in assessment_obj.parts.all()
+    ]
+
+def get_scorer_data(anonymous_scorer_id):
+    """
+    Retrieves the grader's data (full name, username, and email) based on their anonymous ID.
+    """
+    scorer_username = map_anonymized_ids_to_usernames([anonymous_scorer_id]).get(anonymous_scorer_id, "Unknown")
+    scorer_name = map_anonymized_ids_to_fullname([anonymous_scorer_id]).get(anonymous_scorer_id, "Unknown")
+    scorer_email = map_anonymized_ids_to_emails([anonymous_scorer_id]).get(anonymous_scorer_id, "Unknown")
+    return scorer_name, scorer_username, scorer_email
+
+def generate_assessment_data(assessment_list):
+    results = []
+    for assessment in assessment_list:
+
+        scorer_name, scorer_username, scorer_email = get_scorer_data(assessment.scorer_id)
+
+        assessment_data = {
+            "idAssessment": str(assessment.id),
+            "grader_name": scorer_name,
+            "grader_username": scorer_username,
+            "grader_email": scorer_email,
+            "assesmentDate": assessment.scored_at.strftime('%d-%m-%Y'),
+            "assesmentScores": parts_summary(assessment),
+            "problemStep": score_type_to_string(assessment.score_type),
+            "feedback": assessment.feedback or ''
+        }
+
+        results.append(assessment_data)
+    return results
+
+def generate_received_assessment_data(submission_uuid=None):
+    """
+    Generates a list of received assessments data based on the submission UUID.
+
+    Args:
+        submission_uuid (str, optional): The UUID of the submission. Defaults to None.
+
+    Returns:
+        list[dict]: A list containing assessment data dictionaries.
+    """
+
+    results = []
+
+    submission = None
+    if submission_uuid:
+        submission = sub_api.get_submission_and_student(submission_uuid)
+
+    if not submission:
+        return results
+
+    assessments = _use_read_replica(
+        Assessment.objects.prefetch_related('parts').
+        prefetch_related('rubric').
+        filter(
+            submission_uuid=submission['uuid']
+        )
+    )
+    return generate_assessment_data(assessments)
+
+
+def generate_given_assessment_data(item_id=None, submission_uuid=None):
+    """
+    Generates a list of given assessments data based on the submission UUID as scorer.
+
+    Args:
+        submission_uuid (str, optional): The UUID of the submission. Defaults to None.
+
+    Returns:
+        list[dict]: A list containing assessment data dictionaries.
+    """
+    results = []
+    # Getting the scorer student id
+    primary_submission = sub_api.get_submission_and_student(submission_uuid)
```

Comment:
I don't think primary submission says much. Can we be more explicit? I can't think of a name right now, let me know what you come up with!

Reply:
changed it to `scorer_submission` ... it retrieves the `scorer_id` from their `submission_uuid`. To do this, we retrieve the submission object and extract the `student_id` from it.